src/main/java/Duke.java

```diff
@@ -31,49 +32,49 @@ public static void main(String[] args) {
-                    break;
-                case "deadline":
-                    splitCommand = command.split(" ", 2);
-                    date = splitCommand[1].split("/by", 2);
-                    Deadline deadline = new Deadline(date[0].trim(), date[1].trim());
-                    tasks[numberOfTasks] = deadline;
-                    numberOfTasks++;
-                    System.out.println(" Alright, I've added this task:\n   " + deadline.toString() + "\n"
-                            + " Now you have " + numberOfTasks + " tasks in your list.\n" + HORIZONTAL_LINE);
-                    break;
-                case "event":
-                    splitCommand = command.split(" ", 2);
-                    date = splitCommand[1].split("/at", 2);
-                    Event event = new Event(date[0].trim(), date[1].trim());
-                    tasks[numberOfTasks] = event;
-                    numberOfTasks++;
-                    System.out.println(" Alright, I've added this task:\n   " + event.toString() + "\n"
-                            + " Now you have " + numberOfTasks + " tasks in your list.\n" + HORIZONTAL_LINE);
-                    break;
-                case "todo":
-                    toDo = new ToDo(command.substring(5, command.length()));
-                    tasks[numberOfTasks] = toDo;
-                    numberOfTasks++;
-                    System.out.println(" Alright, I've added this task:\n   " + toDo.toString() + "\n"
-                            + " Now you have " + numberOfTasks + " tasks in your list.\n" + HORIZONTAL_LINE);
-                    break;
-                default:
-                    toDo = new ToDo(command);
-                    tasks[numberOfTasks] = toDo;
-                    numberOfTasks++;
-                    System.out.println(" Alright, I've added this task:\n   " + toDo.toString() + "\n"
-                            + " Now you have " + numberOfTasks + " tasks in your list.\n" + HORIZONTAL_LINE);
-                    break;
-                }
+                continue;
+            }
+            String[] tokens = command.split(" ");
+            switch (tokens[0]) {
+            case "done":
+                int taskNumber = Integer.parseInt(tokens[1]);
+                task= tasks[taskNumber-1];
+                task.markAsDone();
+                System.out.println(HORIZONTAL_LINE + "\n Nice! I've marked this task as done:");
+                System.out.println("   " + task.toString()  + "\n" + HORIZONTAL_LINE);
+                break;
+            case "deadline":
+                splitCommand = command.split(" ", 2);
+                date = splitCommand[1].split("/by", 2);
+                Deadline deadline = new Deadline(date[0].trim(), date[1].trim());
+                tasks[numberOfTasks] = deadline;
+                numberOfTasks++;
+                System.out.println(" Alright, I've added this task:\n   " + deadline.toString() + "\n"
```

Comment:
Hey since you repeat the "Alright, I've added this task..." statement for all 3 types of tasks, maybe you want to create a separate method like "printAddedTask" and print the statements from there? you can also update the value of numberOfTasks within the method as well since it is common for all 3 types of tasks:)

Reply:
Thank you for the suggestion! I think it will improve my code readability hence will likely incorporate that in soon :)